oscarapi/middleware.py

```diff
@@ -21,6 +21,28 @@
 HTTP_SESSION_ID_REGEX = re.compile(
     r'^SID:(?P<type>(?:ANON|AUTH)):(?P<realm>.*?):(?P<session_id>.+?)(?:[-:][0-9a-fA-F]+){0,2}$')
 
+LANGUAGE_IN_URL = re.compile(r'(/[a-z][a-z]-[a-z][a-z])(/.*$)')
+
```

Comment:
How about constructing a regex from settings.LANGUAGES or whatever setting it was, which is more precise.